src/components/Weather.js

```diff
@@ -11,24 +12,39 @@ export default class Weather extends Component {
       apiResponse: false,
       temperature: null,
       isCelcius: false,
+      location: false,
+      latitude: null,
+      longitude: null,
     };
   }
 
   componentDidMount() {
-    this.getWeather();
+    if ('geolocation' in navigator) {
+      getLocation().then(res => {
+        this.setState({
+          location: true,
+          latitude: res.coords.latitude,
+          longitude: res.coords.longitude,
+        });
+        this.getWeather(this.state.latitude, this.state.longitude);
```

Comment:
can you `let {latitude, longiture} = this.state` on top of the function to avoid having `this.state` twice?

Reply:
Also why are you sending this down?  I think `getWeather` doesnt get any params?